src/applications/edu-benefits/0994/containers/PreSubmitInfo.jsx

```diff
@@ -8,7 +9,10 @@ function PreSubmitNotice({
   setPreSubmit,
 }) {
   const activeDutyNote = (
-    <div className="vads-u-margin-bottom--3">
+    <div
+      className="vads-u-margin-bottom--3"
+      id={!environment.isProduction() && 'submit-application'}
```

Comment:
Just to make sure I'm understanding this correctly, are you only adding the id here if the environment is not production?

Reply:
Correct, as a feature flag for testing in local dev and Staging.

Reply:
> Just to make sure I'm understanding this correctly, are you only adding the id here if the environment is not production?

Do you know if this is standard operating procedure here? Somewhere along the way we were instructed to dev in this manner (wrap everything in a flag to prevent it from going to prod for uat, then remove wrapper). I'm fairly knew here.

Reply:
Ok cool, I just wanted to make sure I understood what you are doing here. For this particular implementation, the strategy you have here is probably fine. The change is low-risk and it's unlikely that it would work differently in staging vs. production.

But, for future reference, here is the [platform documentation on how we use feature toggles](https://depo-platform-documentation.scrollhelp.site/developer-docs/Feature-toggles.1859780873.html). They're good for testing before a change hits production, but also make it easy to roll back a change if, for whatever reason, it works in staging but not in prod.

Reply:
I'm also concerned that the ID name here isn't specific enough. I don't want to end up in a situation where we have multiple progress buttons being used on the same page that are pointing to different description text.

A better implementation might be to provide a more specific ID and then add a parameter to the ProgressButton for the `ariaDescribedbyId`. Then you can just pass the ID you want to the button when you call it.

Reply:
I did get the naming of the ID from someone in accessibility, but I understand where you're coming from and let me ask you something indirectly related:

It's really not best practice for me to be updating the actual Button component (below, https://github.com/department-of-veterans-affairs/vets-website/pull/19326/files#diff-d51faea197ba71f8b7a2f2f39d4b642cc598bd911e1e62ea00abf3b87b387850). 

What I was trying to do was figure out how to send down a specific prop to the button, and/or create an ariaDescribedbyId tag on that button component to do exactly as you said, but I can't figure out where in the code the button is actually being invoked, nor where the text for that button, in this case: "Submit Application" is specified. I've searched the entire code base and I don't see it, nor is it in content-build. I'm assuming something in a config file invokes the button for this page but I can't see it, would you happen to know this?

Reply:
I don't, unfortunately. But @Mottie has worked on this file recently and may be able to point you in the right direction.

Reply:
No worries. Hey Rob / @Mottie, would appreciate if you have a moment to take a look at this little dilemma :)

Reply:
I don't think checking "Submit Application" would be a good fix. It would only fix your specific application (some apps are not named "application") and the `ProgressButton` component is used _everywhere_.

Also, I don't know if every application has a privacy statement ("By submitting this form"). They all may, but I do know that an ID that doesn't point to anything would also cause an a11y violation.

For reference, the [`SubmitButtons` component](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/review/SubmitButtons.jsx#L9-L15) renders on the review & submit page. It, in turn, renders one of the `submit-status` components which renders the `ProgressButton`. You'd have to update each of those to pass in an `aria-describedby` so it specifically shows up on the review & submit page. It's a lot more convoluted than you think :(

Reply:
Alright Rob, I'm going to wrap my head around what you're saying, but at least that's a lead. So basically it's worse than I thought re: being over-engineered :/

Reply:
@Mottie Alright I'm not understanding then. There is nothing in the 0994 directory where I'm working calling the `ProgressButton` and I'm not sure which Review and Submit page you're referring to? The 0993 form directory does call the `ProgressButton` as a reference (which I was trying to compare to.

This aside, I probably shouldn't be modifying this ProgressButton at all given the scope of my group. I think it would make sense for Platform to modify this button to accept an `aria-describedby` prop, specifically for this reason (and also to be used elsewhere).

Reply:
Sorry, I should have explained better. The review & submit page is rendered by the form system based on the chapters & pages defined within the `config/form.js` file. So we (me included) do not have a lot of control over the rendering of the review & submit page which is built for _all_ forms.

What I was trying to say, in a convoluted way, is that you may have to follow the components all the way back to the review & submit page rendering component and have it pass down the `aria-describedby` to the `ProgressButton`. Start from:

- `src/platform/forms-system/src/js/review/SubmitController.jsx` - Check the `formConfig.formId` for your specific form, then set the `aria-describedby` prop in `SubmitButtons` 
- `src/platform/forms-system/src/js/review/SubmitButtons.jsx` - Pass the `aria-describedby` to each of the submit-status components, which in turn will pass the `aria-describedby` to the `ProgressButton`:
  - `src/platform/forms-system/src/js/review/submit-states/Back.jsx`
  - `src/platform/forms-system/src/js/review/submit-states/ClientError.jsx`
  - `src/platform/forms-system/src/js/review/submit-states/Default.jsx`
  - `src/platform/forms-system/src/js/review/submit-states/Pending.jsx`
  - `src/platform/forms-system/src/js/review/submit-states/Submitted.jsx`
  - `src/platform/forms-system/src/js/review/submit-states/ThrottledError.jsx`
  - `src/platform/forms-system/src/js/review/submit-states/ValidationError.jsx`
- `src/platform/forms-system/src/js/components/ProgressButton.jsx` - Add the `aria-describedby` attribute if passed down

The reason I'm suggesting to do it this way is that it ensures you're only modifying the submit button on the review & submit page for your specific form. `ProgressButton` is used everywhere and it isn't connected to the Redux store, so you can't get the form ID (from `config/form.js`), or the form context (tells you that you're on the review & submit page).

I hope that helps?

And @k80bowman If you see anything that I've suggested that you'd do different, please let me know. Thanks!

Reply:
Thank you for the overview @Mottie! This is really helpful.

Reply:
I've spent some time digging into the layers of all of this and I see what you're saying back up to the `form.js` file. I really appreciate your previous comment, it helped in my understanding quite a bit. 

Couple questions. 

1) Working in the SubmitButtons component, for the case of the `Submit Application` context I'm working on, are any of these other button states relevant? It seems to only be Default. Granted I haven't fully wrapped my head around all the state logic, but thought just adding here and from there downward could work?

```
export default function SubmitButtons(props) {
  const { onBack, onSubmit, submission, formConfig, formErrors = {} } = props;

  const appType = formConfig?.customText?.appType || APP_TYPE_DEFAULT;
  const buttonText =
    formConfig.customText?.submitButtonText || `Submit ${appType}`;

  if (submission.status === false) {
    return (
      <Default
        ariaDescribedBy={"TEST test"}
        buttonText={buttonText}
        onBack={onBack}
        onSubmit={onSubmit}
        formConfig={formConfig}
      />
    );
  } else if (submission.status === 'submitPending') {
    return (
      <Pending onBack={onBack} onSubmit={onSubmit} formConfig={formConfig} />
    );
  } else if (submission.status === 'applicationSubmitted') {
    return (
      <Submitted onBack={onBack} onSubmit={onSubmit} formConfig={formConfig} />
    );
  } else if (submission.status === 'clientError') {
    return (
      <ClientError
        buttonText={buttonText}
        formConfig={formConfig}
        onBack={onBack}
        onSubmit={onSubmit}
      />
    );
```
2) I do feel a bit uncomfortable making changes here with this code being at the platform level which impacts things above and beyond my groups scope. I would think your team would typically make a systemic change like this, or no?

Reply:
1. Yes, the `Default` component is the first one to be rendered, but if there is an error or problem, then the other views won't pass down the `aria-describedby` property to the `ProgressButton`. So it's better to include it in all of the ones I listed (the ones that render a `ProgressButton`)
2. I am also on a VSA team and I feel that even though the code is owned by the VSP/Design system team, they are still open to changes that we make. It's easier to review a change than it is to make it. In the past (and it may be still is), the design team was severely understaffed, so if you wanted this change, you may have to wait a very long time to get it implemented.

Reply:
We are getting into a better place with staffing levels, but for now, yes, if you feel like you need to wait on us to make this change, it is probably not going to happen quickly. We are open to reviewing changes, though.

Reply:
@Mottie - when you have a moment, I have to add some test coverage, but I added changes I think covers what needs to be done. Feedback welcome. I left out the back state as that seemed to be specific to a "Back" button I believe?

Reply:
@k80bowman I've implemented @Mottie 's recommended changes but wanted to circle back on our original discussion. As a team, we're now starting to add actual feature flags as you mentioned. For this particular one just being an aria attribute, if it's ok with you, I'd like to keep any more complexity out of this PR, but obviously the decision rests with you the reviewer.

Reply:
I'm good with this implementation. Thank you for doing all of this work and thank you @Mottie for helping out!